network.go

```diff
@@ -359,17 +359,22 @@ func (n *network) UnmarshalJSON(b []byte) (err error) {
 	}
 	n.name = netMap["name"].(string)
 	n.id = netMap["id"].(string)
-	n.ipamType = netMap["ipamType"].(string)
-	n.addrSpace = netMap["addrSpace"].(string)
 	n.networkType = netMap["networkType"].(string)
 	n.endpointCnt = uint64(netMap["endpointCnt"].(float64))
 	n.enableIPv6 = netMap["enableIPv6"].(bool)
+
 	if v, ok := netMap["generic"]; ok {
 		n.generic = v.(map[string]interface{})
 	}
 	if v, ok := netMap["persist"]; ok {
 		n.persist = v.(bool)
 	}
+	if v, ok := netMap["ipamType"]; ok {
```

Comment:
You are trying here to support the case where an old `network` is cached in the datastore.
If we choose to support that, then don't you have to initialize `n.ipamType = ipamapi.DefaultIPAM` before doing unmarshalling so that old network will use the new ipam plugin. Otherwise allocation would fail.